File: backend/Backend/app/schemas/history.py
import uuid
import datetime
from typing import Optional
from pydantic import BaseModel, Field


class HistoryCreate(BaseModel):
    url: str
    title: Optional[str] = Field(None, max_length=500)
    device_id: Optional[uuid.UUID] = None
    visit_time: Optional[datetime.datetime] = None


class HistoryResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    device_id: Optional[uuid.UUID] = None
    url: str
    title: Optional[str] = None
    visit_time: datetime.datetime
    visit_count: int

    class Config:
        from_attributes = True

File: frontend/lib/core/models/history.dart
class HistoryModel {
  final String id;
  final String url;
  final String? title;
  final DateTime visitTime;
  final int visitCount;

  HistoryModel({
    required this.id,
    required this.url,
    this.title,
    required this.visitTime,
    this.visitCount = 1,
  });

  factory HistoryModel.fromJson(Map<String, dynamic> json) {
    return HistoryModel(
      id: json['id'] as String,
      url: json['url'] as String,
      title: json['title'] as String?,
      visitTime: DateTime.parse(json['visit_time'] as String),
      visitCount: json['visit_count'] as int? ?? 1,
    );
  }
}
